src/tests/activity.test.js

```diff
@@ -1 +1,149 @@
-test.skip('skip', () => {});
+import request from 'supertest';
+import app from '../app';
+import activity from '../models/activity';
+import activitySeed from '../seeds/activity';
+
+const mockActivity = {
+  name: 'Boxing',
+  description: 'vestibulum aliquet ultrices erat tortor sollicitudin mi non mi integer ac neque duis',
+  isActive: false,
+};
+
+const mockActivityExistingName = {
+  name: 'Chandal',
+  description: 'Any description',
+  isActive: false,
+};
+
+const mockLessName = {
+  name: '',
+  description: 'Some text',
+  isActive: true,
+};
+
+const mockLessDescription = {
+  name: 'Name',
+  description: '',
+  isActive: true,
+};
+
+const mockLessIsActive = {
+  name: 'Names',
+  description: 'some description',
+  isActive: '',
+};
+
+let firstActivityId;
+
+beforeEach(async () => {
+  await activity.collection.insertMany(activitySeed);
+  const foundFirstId = await activity.find();
+  if (foundFirstId.length > 0) {
+    // eslint-disable-next-line no-underscore-dangle
+    firstActivityId = foundFirstId[0]._id;
+  }
+});
+
+afterEach(async () => {
+  await activity.collection.deleteMany();
+});
+
+describe('GET /api/activities', () => {
+  test('The database must return 200 if is not empty of activities', async () => {
+    const response = await request(app).get('/api/activities').send();
+    expect(response.body.data.length).toBeGreaterThan(0);
+    expect(response.status).toBe(200);
+    expect(response.body.error).toBeFalsy();
+    expect(response.body).toBeDefined();
+    expect(response.body.message).toMatch('Complete activities list');
+  });
+  test('The status must be 404, if the url is wrong', async () => {
+    const response = await request(app).get('/activities').send();
+    expect(response.status).toBe(404);
+    expect(response.body.error).toBeUndefined();
+  });
+  describe('GET /api/activities', () => {
+    test('The database must return 404 if it\'s empty of activities. An said There are no activities', async () => {
+      await activity.collection.deleteMany({});
+      const response = await request(app).get('/api/activities').send();
+      expect(response.body.data).toBe(undefined);
+      expect(response.status).toBe(404);
+      expect(response.body.message).toMatch(/There are no activities/);
+    });
+  });
+});
+
+describe('GET BY ID /api/activities/:id', () => {
+  test('The database must return 200 if the id exist', async () => {
+    const response = await request(app).get(`/api/activities/${firstActivityId}`).send();
+    expect(response.status).toBe(200);
+    expect(response.body).toBeDefined();
+    expect(response.body.data).toHaveProperty('_id');
+    expect(response.body.data).toHaveProperty('name');
+    expect(response.body.data).toHaveProperty('description');
+    expect(response.body.data).toHaveProperty('isActive');
```

Comment:
To save some lines of code, you could use a single .toHaveProperty() and list all the properties you want to evaluate within it, separated by commas. 
```suggestion
    expect(response.body.data).toHaveProperty('_id', 'name', 'description', 'isActive');
```

Reply:
Great, i didn't know i can do this. Thanks!